datafusion/core/src/physical_plan/filter.rs

```diff
@@ -153,7 +154,19 @@ impl ExecutionPlan for FilterExec {
     }
 
     fn ordering_equivalence_properties(&self) -> OrderingEquivalenceProperties {
-        self.input.ordering_equivalence_properties()
+        let stats = self.statistics();
+        // Add the columns that have only one value (singleton) after filtering to constants.
```

Comment:
Why is there a a difference between `OrderingEquivalenceProperties` and  `EquivalenceProperties` -- it seems like using statistics based equivalence as well as predicate based equivalence would be relevant to both 



In other words, if the filter has a predicate like `column = 5` shouldn't `column` be added to list of constants even if the `column` had more than one value in  the statistics?

Reply:
> In other words, if the filter has a predicate like column = 5 shouldn't column be added to list of constants even if the column had more than one value in the statistics?
In this case, statistics should be able to determine `column` will have single value(5) onwards. Hence I presumed, using statistics is sufficient. However, if there are cases, where it is obvious from the predicate value is constant, but statistics fail to resolve it. We can change this implementation I think